subgraph/src/mappings/fill-manager.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  RequestFilled as RequestFilledEvent
+} from "../../generated/FillManager/FillManager"
+import { Fill } from "../../generated/schema";
+import { Address, Bytes } from "@graphprotocol/graph-ts";
+import { TRANSFER_SIGNATURE } from "../utils/constants";
+
+import { loadTransaction } from "../utils";
+export function handleRequestFilled(event: RequestFilledEvent): void {
+  let fill = new Fill(
+    event.transaction.hash.concat(Bytes.fromUTF8('-' + event.logIndex.toString()))
+  )
+
+  fill.requestId = event.params.requestId
+  fill.sourceChainId = event.params.sourceChainId
+  fill.targetTokenAddress = event.params.targetTokenAddress
+  fill.filler = event.params.filler
+  fill.amount = event.params.amount
+  fill.transaction = loadTransaction(event).id
+  fill.emittedAt = event.block.timestamp
+  fill.emittedByTx = event.transaction.hash
+  fill.contractAddress = event.address
+  const receipt = event.receipt
+
+  // console.log('no receipt')
+  if (receipt) {
+    for (let index = 0; index < receipt.logs.length; index++) {
```

Comment:
I actually tried to use this when I was writing the subgraph, but I ran into this:
![image](https://github.com/beamer-bridge/beamer/assets/693770/ac70b6b2-3862-4d24-b89d-b6c6dd16b03f)

TS2488: Type 'Log[]' must have a '[Symbol.iterator]()' method that returns an iterator.

```
  export class TransactionReceipt {
    constructor(
      public transactionHash: Bytes,
      public transactionIndex: BigInt,
      public blockHash: Bytes,
      public blockNumber: BigInt,
      public cumulativeGasUsed: BigInt,
      public gasUsed: BigInt,
      public contractAddress: Address,
      public logs: Array<Log>,
      public status: BigInt,
      public root: Bytes,
      public logsBloom: Bytes,
    ) {}
  }
```
The receipt logs definition is an array<Log> which should satisfy the for ... of loop, but for some reason typescript is not happy. So, since the conventional loop was working I didn't want to go through the trouble of debugging stupid typescript issues...